### PR TITLE
Only log cljfmt+clojure-lsp error if clojure-lsp has started

### LINKED ISSUE #2078: cljfmt isn't found unless project is started
`Fetching formatting settings from clojure-lsp failed. Check that you are running a version of clojure-lsp that provides "cljfmt-raw" in serverInfo.`

If I launch vscode I get the above message, and cljfmt wont format correctly.  If I then start my project (lein repl), which happens to include lein-cljfmt, then cljfmt starts formatting correctly.  More interestingly, if I stop the project but don't close vscode, cljfmt continues to format correctly.  

This is a new behavior as of a few days (I was off 2/4 - 2/9)

Im not sure where clojure-lsp is looking for cljfmt, is there somewhere else besides project.clj / plugins I need to be updating?

I think cljfmt had a pretty big changed merged in recently, but can't confirm the timeline.

### PR DESCRIPTION
**Pull request by PEZ**
## What has changed?

- Log error to console instead of alerting about it
- Only log error if the server has started

The trade-off here is that the user might get default formatting without having been noticed. But it should be rare, and not the worlds biggest deal.

* Fixes #2078

## My Calva PR Checklist

I have:

- [x] Read [How to Contribute](https://github.com/BetterThanTomorrow/calva/wiki/How-to-Contribute#before-sending-pull-requests).
- [x] Directed this pull request at the `dev` branch. (Or have specific reasons to target some other branch.)
- [x] Made sure I have changed the PR base branch, so that it is not `published`. (Sorry for the nagging.)
- [x] Updated the `[Unreleased]` entry in `CHANGELOG.md`, linking the issue(s) that the PR is addressing.
- [x] Figured if **anything** about the fix warrants tests on Mac/Linux/Windows/Remote/Whatever, and either tested it there if so, or mentioned it in the PR.
- ~[ ] Added to or updated docs in this branch, if appropriate~
- [ ] Tests
  - [ ] Tested the particular change
  - [ ] Figured if the change might have some side effects and tested those as well.
- [x] Referenced the issue I am fixing/addressing _in a commit message for the pull request_.
  - [x] If I am fixing the issue, I have used [GitHub's fixes/closes syntax](https://help.github.com/en/articles/closing-issues-using-keywords)
- [x] Formatted all JavaScript and TypeScript code that was changed. (use the [prettier extension](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) or run `npm run prettier-format`)
- [x] Confirmed that there are no linter warnings or errors (use the [eslint extension](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint), run `npm run eslint` before creating your PR, or run `npm run eslint-watch` to eslint as you go).

Ping @pez, @bpringe, @corasaurus-hex, @Cyrik

### CODE REVIEW

File: src/formatter-config.ts [hunk 1]
```diff
@@ -51,15 +51,14 @@ export async function getConfig(
...
+    if (client && client.isRunning) {
       lspFormatConfig = await lsp.api.getCljFmtConfig(client);
     }
-  }
-  if (configPath === LSP_CONFIG_KEY && !lspFormatConfig) {
-    void vscode.window.showErrorMessage(
-      'Fetching formatting settings from clojure-lsp failed. Check that you are running a version of clojure-lsp that provides "cljfmt-raw" in serverInfo.',
-      'Roger that'
-    );
+    if (client.isRunning && !lspFormatConfig) {
```
Comment (bpringe) [inline src/formatter-config.ts]: Would this error get logged even if `configPath !== LSP_CONFIG_KEY`? Does that mean this error will get logged always by default, until someone sets `configPath === LSP_CONFIG_KEY`?

Comment (bpringe) [reply src/formatter-config.ts]: On another note, I think we need a check here for the client existing like you added in the `if` condition above -- `client && client.isRunning`. See [this comment](https://github.com/BetterThanTomorrow/calva/issues/2078#issuecomment-1518326258). Better yet though, you can nest the check for `lspFormatConfig` in the first `if` block. Something like:

```javascript
if (client && client.isRunning && configPath === LSP_CONFIG_KEY) { // see question above on that last check
  lspFormatConfig = await lsp.api.getCljFmtConfig(client);
  if (!lspFormatConfig) {
    console.error(...
  }
```

Comment (PEZ) [reply src/formatter-config.ts]: Not sure I understand the first concern. It is all protected with a check for `configPath === LSP_CONFIG_KEY`, right?

I fixed the second concern in the latest commit. Please have a look. Maybe your first concern still needs to be addressed.

Comment (bpringe) [reply src/formatter-config.ts]: Oh, I didn't see that check before. Never mind the first concern.

**Pull request merged**